src/Data/Foldable1.hs

```diff
@@ -615,6 +621,44 @@ instance Foldable1 (Tagged b) where
     maximum = coerce
 #endif
 
+-------------------------------------------------------------------------------
+-- ghc-prim
+-------------------------------------------------------------------------------
+
+#ifdef MIN_VERSION_ghc_prim
+#if MIN_VERSION_ghc_prim(0,10,0)
+instance Foldable1 Solo where
```

Comment:
Let's not merge this. The `0.10.0` case should happen to begin with, as `Data.Foldable1` is in `base` then.

Reply:
Ah, you are quite right. I've removed the `MIN_VERSION_ghc_prim(0,10,0)` parts.

As far as GHC 9.6 support goes, other things will need to happen besides CI changes before this library fully supports 9.6. I will defer that to a separate PR.